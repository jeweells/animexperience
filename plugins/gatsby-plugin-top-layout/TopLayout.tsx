import * as React from 'react'
import PropTypes from 'prop-types'
import CssBaseline from '@mui/material/CssBaseline'
import { ThemeProvider } from '@mui/material/styles'
import theme from '../../src/theme'
import { PropsWithChildren } from 'react'

export default function TopLayout(props: PropsWithChildren<{}>) {
    return (
        <React.Fragment>
            <ThemeProvider theme={theme}>
                {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
                <CssBaseline />
                {props.children}
            </ThemeProvider>
        </React.Fragment>
    )
}

TopLayout.propTypes = {
    children: PropTypes.node,
}
