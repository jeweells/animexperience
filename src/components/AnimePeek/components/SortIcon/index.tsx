import React from 'react'
import SvgIcon from '@mui/material/SvgIcon'
import styled from 'styled-components'

export type SortIconProps = {
    order: 'asc' | 'desc'
    size: number
    spacing?: number
    stroke?: number
}

const Svg = styled(SvgIcon)`
    * {
        transition: all 300ms ease-in-out;
    }
`

export const SortIcon = React.memo<SortIconProps>(
    ({ order, size, spacing = 5, stroke = 3 }) => {
        return (
            <Svg>
                <line
                    x1={0}
                    x2={size}
                    y2={size / 2 - stroke - spacing}
                    y1={size / 2 - stroke - spacing}
                    strokeWidth={stroke}
                    stroke={'currentColor'}
                    style={{
                        transform: `scaleX(${order === 'asc' ? 0.25 : 1})`,
                    }}
                />
                <line
                    x1={0}
                    x2={size}
                    y2={size / 2}
                    y1={size / 2}
                    strokeWidth={stroke}
                    stroke={'currentColor'}
                    style={{
                        transform: 'scaleX(0.5)',
                    }}
                />
                <line
                    x1={0}
                    x2={size}
                    y2={size / 2 + stroke + spacing}
                    y1={size / 2 + stroke + spacing}
                    strokeWidth={stroke}
                    stroke={'currentColor'}
                    style={{
                        transform: `scaleX(${order === 'desc' ? 0.25 : 1})`,
                    }}
                />
            </Svg>
        )
    },
)

SortIcon.displayName = 'SortIcon'

export default SortIcon
