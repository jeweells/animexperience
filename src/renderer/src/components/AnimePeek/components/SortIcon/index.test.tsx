import SortIcon, { SortIconProps } from './index'
import TopLayout from '../../../../../plugins/gatsby-plugin-top-layout/TopLayout'

const orders: Array<SortIconProps['order']> = ['asc', 'desc']

describe('SortIcon', () => {
  let props: SortIconProps

  const getComponent = (override: Partial<SortIconProps> = {}) => {
    return (
      <TopLayout>
        <SortIcon {...props} {...override} />
      </TopLayout>
    )
  }

  beforeEach(() => {
    props = {
      order: 'asc',
      size: 30
    }
  })

  it.each(orders.map((order) => [order]))('renders default with order=%s', async (order) => {
    const wrapper = render(getComponent({ order: order as SortIconProps['order'] }))
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
