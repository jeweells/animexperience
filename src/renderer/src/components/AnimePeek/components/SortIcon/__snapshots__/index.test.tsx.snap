// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortIcon renders default with order=asc 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

.c1 * {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

<svg
    aria-hidden="true"
    class="c0 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c1"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <line
      stroke="currentColor"
      stroke-width="3"
      style="transform: scaleX(0.25);"
      x1="0"
      x2="30"
      y1="7"
      y2="7"
    />
    <line
      stroke="currentColor"
      stroke-width="3"
      style="transform: scaleX(0.5);"
      x1="0"
      x2="30"
      y1="15"
      y2="15"
    />
    <line
      stroke="currentColor"
      stroke-width="3"
      style="transform: scaleX(1);"
      x1="0"
      x2="30"
      y1="23"
      y2="23"
    />
  </svg>
</DocumentFragment>
`;

exports[`SortIcon renders default with order=desc 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

.c1 * {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

<svg
    aria-hidden="true"
    class="c0 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c1"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <line
      stroke="currentColor"
      stroke-width="3"
      style="transform: scaleX(1);"
      x1="0"
      x2="30"
      y1="7"
      y2="7"
    />
    <line
      stroke="currentColor"
      stroke-width="3"
      style="transform: scaleX(0.5);"
      x1="0"
      x2="30"
      y1="15"
      y2="15"
    />
    <line
      stroke="currentColor"
      stroke-width="3"
      style="transform: scaleX(0.25);"
      x1="0"
      x2="30"
      y1="23"
      y2="23"
    />
  </svg>
</DocumentFragment>
`;
